doc-site/docs/architecture/distributed_sequencer.md

```diff
@@ -0,0 +1,1437 @@
+
+Note that there is a brief dip in throughput as node B waits for node A to flush through the pending dispatched transactions and there is also some additional processing for the inflight transactions that haven't been dispatched yet.  So a range size of 4 is unreasonable and it would be more likely for range sizes to be much larger so that these dips in throughput become negligible.
+
+TODO
+
+ - need more detail on precisely _how_  nodes B, C and D know that transactions `B-2``A-2` , `B-2`, `C-2` and `D-2` are past the point of no return and that transactions `A-3` , `B-3`, `C-3` and `D-3` do need to be re-delegated.
+ - need more detail on precisely _how_ node B can continue to coordinate the assembly of transactions `A-3` , `B-3`, `C-3` and `D-3` in a domain context that is aware of the speculative states locks from transactions `B-2``A-2` , `B-2`, `C-2` and `D-2` 
+ - It might be more productive for the next level of detail on these points to come in the form of a proposal in code.
+
+### Variation in block height
+It is likely that different nodes will become aware of new block heights at different times so the algorithm must accommodate that inconsistency. 
+
+ - given that different nodes index the blockchain events with varying latency, it is not assumed that all nodes have the same awareness of "current block number" at any one time. This is accommodated by the following
+ - Each node delegates all transactions to which ever node it determines as the current coordinator based on its latest knowledge of "current block"
+ - The delegate node will accept the delegation if its awareness of current block also results in it being chosen by the selector function.  Otherwise, the delegate node rejects the delegation and includes its view of current block number in the response
+ - On receiving the delegation rejection, the sender node can determine if it is ahead or behind (in terms of block indexing) the node it had chosen as delegate.  
+ - If the sender node is ahead, it continues to retry the delegation until the delegate node finally catches up and accepts the delegation
+ - If the sender node is behind, it waits until its block indexer catches up and then selects the coordinator for the new range
+ - Coordinator node will continue to coordinate ( send endorsement requests and submit endorsed transactions to base ledger) until its block indexer has reached a block number that causes the coordinator selector to select a different node.
+ - at that time, it waits until all dispatched transactions are confirmed on chain, then delegates all current inflight transactions to the new coordinator.
+ - if the new coordinator is not up to date with block indexing, then it will reject and the delegation will be retried until it catches up. 
+ - while a node is the current selected coordinator, it sends endorsement requests to every other node for every transaction that it is coordinating
```

Comment:
This seems like an important part of the algorithm for sure, but I'm not sure why the spec for it is under `variation in block height`